core/src/main/java/com/graphhopper/routing/util/NameSimilarityEdgeFilter.java

```diff
@@ -46,34 +45,61 @@
  */
 public class NameSimilarityEdgeFilter implements EdgeFilter {
 
-    private static final Pattern NON_WORD_CHAR = Pattern.compile("[^\\p{L}]+");
-    private final double JARO_WINKLER_ACCEPT_FACTOR = .79;
-    private final JaroWinkler jaroWinkler = new JaroWinkler();
 
+    private static final Map<String, String> DEFAULT_REWRITE_MAP = new HashMap<String, String>() {{
+        // two char words will be ignored but ignore certain longer phrases (or rename them)
+        for (String remove : Arrays.asList(
+                "ally", "alley",
+                "arc", "arcade",
+                "bvd", "boulevard",
+                "av.", "avenue", "avenida",
+                "calle",
+                "rd.", "road",
+                "ln.", "lane",
+                "pl.", "place", "plaza",
+                "str.", "str", "straße", "strasse", "st.", "street", "strada",
+                "sq.", "square",
+                "tr.", "track",
+                "via")) {
+            put(remove, "");
```

Comment:
Do you think this is now sufficient and we could add new stuff if required later? I think adding all these words does not necessarily improve the result. We probably could try an automated way to find areas where our snapping is not optimal now. E.g. pick a location and 4 close streets and pick one street to see if the others get rejected or similar.

Reply:
Ok sure, we can always add more later and see how we go :). 

Personally, I would add:
crescend, cres, cres., parade, pde, pde., boulevard, bvd, bvd.

These are really common ones. In GB there are many Close, Cl, which could make sense to add as well. Probably we just need to see how we go.

Reply:
Have added a few as requested :)